Duplicati/Library/Utility/CallContextSettings.cs

```diff
@@ -277,11 +277,11 @@ public static T Settings
         /// </summary>
         /// <returns>The disposable handle for the context.</returns>
         /// <param name="initial">The initial value.</param>
-        public static IDisposable StartContext(string settingsKey, T initial = default(T))
+        public static IDisposable StartContext(string settingsKey, T? initial = default(T))
         {
             contextSettingsType = settingsKey;
             var res = new ContextGuard();
-            Settings = initial;
+            Settings = initial!;

```

Comment:
Can this be done better? If we allow it to be null, then `Settings` should be nullable as well?